### DISCUSSION TITLE
Supports maximum text length

### DISCUSSION THREAD

**Discussion by tianchiguaixia** [explosion/spacy-llm#399] [Q&A]
Regarding spacy_llm, what is the maximum text length it supports? If the text length in a single instance is very long, exceeding the limit of OpenAI, how does it handle this?

**Comment by rmitsch** [comment]
Hi @tianchiguaixia, currently `spacy-llm` doesn't do anything about overly long texts. We are currently finishing a [feature](https://github.com/explosion/spacy-llm/pull/332) though that will use a map-reduction approach to handling texts that are too long for models' context windows. In a nutshell, docs will be split in as many shards as necessary, each shard will be processed by the LLM independently, and finally all shards will be merged back into a single `Doc` instance.

This will be included in v1.0.0 of `spacy-llm`, which we hope to release in late December or early January.
